src/modules/pos/validator/components/RegisterValidatorView/index.js

```diff
@@ -7,19 +7,27 @@ import Summary from '../RegisterValidatorSummary';
 import Status from '../RegisterValidatorStatus';
 import styles from './registerValidator.css';
 
-const RegisterValidator = ({ history, t }) => (
-  <Dialog hasClose>
-    <MultiStep
-      className={styles.multiStep}
-      prevPage={history.goBack}
-      backButtonLabel={t('Back')}
-    >
-      <Form t={t} />
-      <Summary t={t} />
-      <TxSignatureCollector t={t} />
-      <Status t={t} />
-    </MultiStep>
-  </Dialog>
-);
+const RegisterValidator = ({ history, t }) => {
+  const [isStepTxSignatureCollector, setIsStepTxSignatureCollector] = useState(false);
+  const onMultiStepChange = useCallback(({ step: { current } }) => {
```

Comment:
Similar to `useMemo` in the case that our MultiStep component is not optimised enough to make this a performance optimisation it will instead lead to wore performance and more complicated code.

Reply:
The reason for this is because navigating between steps would cause a re-render every time since we want to track the step the `multistep` component has taken but all we need is a function that wouldn't get re-instantiated every time this occurs the reason, why I strongly advise we have a useCallback wrapping this function. 

Assuming the number of renders done by the host component isn't affected by stepping through the children then just  having a function would have been preferred. @oskarleonard @soroushm

Reply:
I understand that this is what we wanted to achieve. However, adding a callback here will not prevent that and It would still re-render as many times as if we dont have a callback here.

Reply:
@eniolam1000752 if the function in children is not a dependencies is not helping any how, unlike useMemo useCallback will memoize by each render that means its increase the memory usage

Reply:
by the way i think it can be simplify by 
```js
const dialogSize = useMemo(
() => [2, 3].includes(current) ? 'sm' : undefiend
, [current]);
```

```
<Dialog hasClose size={dialogSize}>
```